src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -372,11 +372,11 @@ protected void retrieve(SCMSourceCriteria criteria, @NonNull SCMHeadObserver obs
                                         public SCMSourceCriteria.Probe create(@NonNull MergeRequestSCMHead head,
                                                                               @Nullable MergeRequestSCMRevision revision)
                                                 throws IOException, InterruptedException {
-                                            boolean trusted = request.isTrusted(head);
-                                            if (!trusted) {
+                                            isTrusted = request.isTrusted(head);
```

Comment:
I assume this call's the right strategy inside `ForkMergeRequestDiscoveryTrait` and calls `checkTrusted` on that strategy?

Reply:
Yes this one works and does exactly that. But previously the request that we made with the traits inside `getTrustedRevision` doesn't work for fork MRs (although it returns correctly for origin MRs). Idk what is wrong with it though. Anyway even after getting the whether the head is trusted or not why call it again. In addition to that these changes work as expected.